src/main/scala/euler/archives/page/one/problemOne.scala

```diff
@@ -0,0 +1,217 @@
+
+  }
+
+
+
+
+  def largestProductInGrid(grid: Array[Array[Int]]=problemOne.grid) : Int = {
+    var (c1,c2,c3,c4,r1,r2,r3,r4, max) = (0,0,0,0,0,0,0,0,Integer.MIN_VALUE)
+    var check = 0;
+    for{row <- grid.indices
+        column <- grid.indices} {
+      // while on the current column we want to go forward by three row+
+
+      if(row +3 < grid.length){
+//        check the product of the grid column and row
+        check = grid(column)(row)* grid(column)(row+1)*grid(column)(row+2)*grid(column)(row+3)
+        if(check>max){
+          max = check
+//          saving the indices for use later
+          r1 = row
+          r2 = row+1
+          r3 = row + 2
+          r4 = row + 3
+          c1 = column
+          c2 = column
+          c3 = column
+          c4 = column
+        }
+      }
+//      while on the current row we want to go three column forward
+      if(column + 3 < grid.length){
+//        check the product of the grid column and rows
+        check = grid(column)(row) * grid(column+1)(row)* grid(column+2)(row) * grid(column+3)(row)
+        if(check > max){
+            max = check
+          //          saving the indices for use later
+          r1 = row
+          r2 = row
+          r3 = row
+          r4 = row
+          c1 = column
+          c2 = column +1
+          c3 = column + 2
+          c4 = column + 3
+        }
+      }
+
+//      we want to go diagonally for row and column, meaning as we move one row, we would move one column, this would produce a diagonal like structure
+      if((row +3  < grid.length) && (column +3 < grid.length)){
+        //        check the product along the diagonal downward
+
+        check = grid(column)(row) * grid(column+1)(row+1)* grid(column+2)(row+2) * grid(column+3)(row+3)
+        if(check > max){
+          max = check
+          r1 = row
+          r2 = row + 1
+          r3 = row + 2
+          r4 = row + 3
+          c1 = column
+          c2 = column +1
+          c3 = column + 2
+          c4 = column + 3
+        }
+      }
+    // we want to go diagonally for row and column, meaning as we move one row,
+    // we would move one column, this would produce a diagonal like structure but this time around upward
+      if(row > 3 && (column +3 < grid.length)){
+        //        check the product along the diagonal upward
+        check =  grid(column)(row) * grid(column +1)(row -1) * grid(column+2)(row-2) * grid(column +3)(row-3)
+        if(check > max ){
+          max = check
+          r1 = row
+          r2 = row - 1
+          r3 = row - 2
+          r4 = row - 3
+          c1 = column
+          c2 = column +1
+          c3 = column + 2
+          c4 = column + 3
+        }
+      }
+
+    }
+
+
+    max
+  }
+
+  def totalNameScores:Int = {
+    import scala.io.Source
+    val bufferedSource = Source.fromFile("p022_names.txt")
+    var text: String = ""
+    for (line <- bufferedSource.getLines()) {
+      text += line
+    }
+    bufferedSource.close
```

Comment:
This can be simplified, can't it?

`val text = Source.fromFile("p022_names.txt").getLines.mkString` 

should do the trick - you don't need to copy to the buffer (or am i missing something?)

Reply:
Yes it can be, but I was trying to close the the bufferedSource after reading is complete

Reply:
in which case you can do 

```
val source = scala.io.Source.fromFile("file.txt")
val lines = try source.mkString finally source.close()
```

which will release the resource and remain more idiomatically Scala